files/image_config/process-reboot-cause/process-reboot-cause

```diff
@@ -115,9 +144,8 @@ def main():
                  previous_reboot_cause = UNKNOWN_REBOOT_CAUSE
             os.remove(FIRST_BOOT_PLATFORM_FILE)
     # Write the previous reboot cause to PREVIOUS_REBOOT_CAUSE_FILE
-    prev_cause_file = open(PREVIOUS_REBOOT_CAUSE_FILE, "w")
-    prev_cause_file.write(previous_reboot_cause)
-    prev_cause_file.close()
+    with open(PREVIOUS_REBOOT_CAUSE_FILE, "w") as prev_cause_file:
+        prev_cause_file.write(previous_reboot_cause)
```

Comment:
This is not an issue for this PR or change request for this PR. Just want to start the discussion here.

This change is the result of a team brainstorm of various combinations of hard/soft reboot reason. We fell that we've covered all the cases for now. But in case we didn't. We could consider change the previous reboot cause file to 3 lines:

Reboot cause: <calculation result>
Detected hardware cause: <hardware cause>
Detected software cause: <software cause>

Not sure if any other part of our code has dependency on this file being single line?

Reply:
personally I agree your point. 
in addition, if reboot causes from different sources are conflict, it's better to have it in a detailed way. 
meanwhile, the drawback is it can confuse user sometimes. for example, if the following info displayed:

    calculated reboot cause: warm reboot
    hardware cause: power off
    software cause: warm reboot

probably it's better to only display in a 3-line way in case of we known for sure there is conflicts among causes from different sources?

Reply:
You got a good point. However, if we know there is a conflict, then we would have figured it out and fixed it in code. We cannot predict what we don't know yet. :-)